File: py/Komputer.py
# Komputer.py
class Komputer:
    def __init__(self, nama="", cpu=None, ramList=None, harddrive=None, peripheral=None):
        self.nama = nama
        self.cpu = cpu if cpu is not None else Cpu()
        self.ramList = ramList if ramList is not None else []
        self.harddrive = harddrive if harddrive is not None else Harddrive()
        self.peripheral = peripheral if peripheral is not None else Peripheral()
    
    def setNama(self, nama):
        self.nama = nama
    
    def setCpu(self, cpu):
        self.cpu = cpu
    
    def setRam(self, ramList):
        self.ramList = ramList
    
    def setHarddrive(self, harddrive):
        self.harddrive = harddrive
    
    def setPeripheral(self, peripheral):
        self.peripheral = peripheral
    
    def addRam(self, ram):
        self.ramList.append(ram)
    
    def getNama(self):
        return self.nama
    
    def getCpu(self):
        return self.cpu
    
    def getRamList(self):
        return self.ramList
    
    def getHarddrive(self):
        return self.harddrive
    
    def getPeripheral(self):
        return self.peripheral

File: cpp/Komputer.cpp
#pragma once
#include <iostream>
#include <string>
#include <vector>
#include "Cpu.cpp"
#include "Harddrive.cpp"
#include "Ram.cpp"


using namespace std;


class Komputer
{
private:
    string nama;
    Cpu cpu;
    vector<Ram> ramList;
    Harddrive harddrive;
    Peripheral peripheral;
public:
    Komputer()
    {


    }


    Komputer(string nama, Cpu cpu, vector<Ram> ramList, Harddrive harddrive, Peripheral peripheral)
    {
        this->nama = nama;
        this->cpu = cpu;
        this->ramList = ramList;
        this->harddrive = harddrive;
        this->peripheral = peripheral;
    }

    void setNama(string nama)
    {
        this->nama = nama;
    }

    void setCpu(Cpu cpu)
    {
        this->cpu = cpu;
    }


    void setRam(vector<Ram> ramList)
    {
        this->ramList = ramList;
    }    
   
    void setHarddrive(Harddrive harddrive)
    {
        this->harddrive = harddrive;
    }

    void setPeripheral(Peripheral peripheral)
    {
        this->peripheral = peripheral;
    }


    void addRam(Ram ram)
    {
        this->ramList.push_back(ram);
    }


    string getNama()
    {
        return this->nama;
    }


    Cpu getCpu()
    {
        return this->cpu;
    }


    vector<Ram> getRamList() // return vector
    {
        return this->ramList;
    }


    Harddrive getHarddrive()
    {
        return this->harddrive;
    }

    Peripheral getPeripheral()
    {
        return this->peripheral;
    }

    ~Komputer()
    {


    }
};
